Data browser component function re-build

ddjnw1yu opened the pull request:
- Using $emit to return URL query data from sub-component, update URL in one place.
- Adding compare function to remove duplicate URL update/data fetch.

ddjnw1yu committed:
update fetch data logic

ddjnw1yu committed:
Merge branch 'dev' into update-access-fetch

ddjnw1yu committed:
Merge branch 'dev' into update-access-fetch

ddjnw1yu committed:
update fetch logic

ddjnw1yu committed:
re-write filter component logic

ddjnw1yu committed:
move fetch to FilterData

ddjnw1yu committed:
update relation change

ddjnw1yu committed:
update naming

ddjnw1yu committed:
move updateURL to DataContainer

ddjnw1yu committed:
add browser back/forward logic

ddjnw1yu committed:
update convertFacets()

ddjnw1yu committed:
Merge branch 'dev' into data-broswer-components-rebuild

ddjnw1yu committed:
update search component

ddjnw1yu committed:
fix clear search not loading issue

ddjnw1yu committed:
update page/limit component

ddjnw1yu committed:
fix fetch issue when refresh page with facet in url

ddjnw1yu committed:
update fetch when change page

ddjnw1yu committed:
Merge branch 'dev' into data-broswer-components-rebuild

ddjnw1yu committed:
Merge branch 'dev' into data-broswer-components-rebuild

ddjnw1yu committed:
update url when invalid facet used

ddjnw1yu committed:
fix pagenation after refresh

ddjnw1yu committed:
Merge branch 'dev' into data-broswer-components-rebuild

ddjnw1yu committed:
pagination sync

ddjnw1yu committed:
update order logic

ddjnw1yu committed:
handle update url query function

ddjnw1yu committed:
wrong facet in url

alan-wu commented on components/DataBrowser/DataContainer.vue:
```diff
@@ -58,105 +72,193 @@
-      relation: 'and',
-      sortBy: 'Published(asc)',
-    }
+      pageNumber: undefined,
```
It is generally better to not change the type of the value, use negative integer for undefined numeric value will be better.

alan-wu commented on components/DataBrowser/DataContainer.vue:
```diff
@@ -58,105 +72,193 @@
+      facetList: [],
+      filterDict: {},
+      searchContent: "",
+      relationType: true,
```
Rename this variable to something else, relationType == true does not tell us that it is "and".

alan-wu commented on components/DataBrowser/DataContainer.vue:
```diff
@@ -58,105 +72,193 @@
-        this.allFilterDict = await backendQuery.fetchFilterData(this.$config.query_api_url, false);
+    compareFilter(oldFilter, newFilter) {
+      let isDifferent = false;
+      if (Object.keys(newFilter).length !== Object.keys(oldFilter).length) {
```
Consider putting the lengths in a variable in this functions as the Object.keys are called in multiple places.

alan-wu commented on components/DataBrowser/FilterData.vue:
```diff
@@ -73,23 +73,26 @@
+      selectedFacetList: [],
+      selectedFilterDict: {},
+      filterDictResult: {},
       relation: true,
```
A better name for this variable will be great too.

alan-wu reviewed (changes requested):
It is good to refactor these codes, thanks. I have few suggestions. I will test out the changes later.

ddjnw1yu committed:
change request

1. set undefined numeric to -1
2. update filter compare function
3. renaming variable

alan-wu reviewed (approved):
Looks good and the find datasets page seem to work faster.

ddjnw1yu committed:
Merge branch 'dev' into data-broswer-components-rebuild
